### DISCUSSION TITLE
Question about the new userScript API.

### DISCUSSION THREAD

**Discussion by smordarski** [uBlockOrigin/uBlock-issues#2974] [Q&A] [answered]
Will the the [new userScript API](https://developer.chrome.com/blog/chrome-120-beta-whats-new-for-extensions/#new-userscripts-api) help uBlock Origin with manifest v3 to fetch filter lists? I know that uBOL can't fetch external filter lists off the internet and I am wondering if the userScript API will improve uBO on manifest v3. Here's a direct quote from Google about the userScript API:

> User scripts are (usually relatively small) snippets of code that extensions can inject into web pages in order to modify the page's appearance or behavior. They can be created directly by the user or discovered in a number of different user script repositories around the web.

There is more information about this from these links:
https://developer.chrome.com/blog/resuming-the-transition-to-mv3/
https://developer.chrome.com/docs/extensions/reference/userScripts/

This link talks about improvements to content filtering in manifest v3:
https://developer.chrome.com/blog/improvements-to-content-filtering-in-manifest-v3/

**Comment by gorhill** [comment, marked as answer]
uBO Lite uses MV3 [scripting API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting) to inject scriptlets and it's working well, there is no need for that [userScript API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/userScripts).
